deployments/vm/lib-vm.sh

```
@@ -506,10 +514,43 @@ caddyfile() {
+    # ask_secret, when non-empty, is amp-api's own auto-generated
+    # thunder-ask-secret (agent-manager-service/config: THUNDER_ASK_SECRET) —
+    # install-vm.sh reads it back out of the already-deployed amp Secret after
+    # the "amp" Helm release exists (start_caddy runs after run_install's base
+    # installer, so it's always there by this point) and passes it in here.
+    # Empty when unset/unreadable (e.g. an existingSecret override with no
+    # matching key) — the ask call still works either way, it just draws from
+    # AMS's shared public rate-limit budget instead of its own, same as before
+    # this header existed.
+    local ask_secret_header=""
+    [[ -n "$ask_secret" ]] && ask_secret_header=$'\t\theader_up X-Thunder-Ask-Secret '"${ask_secret}"$'\n'
+    printf 'http://127.0.0.1:9753 {\n\t@gateway_or_agents expression `{query.domain}.endsWith(".%s") || {query.domain}.endsWith(".%s")`\n\trespond @gateway_or_agents 200\n\treverse_proxy 127.0.0.1:8080 {\n\t\theader_up Host %s\n%s\t}\n}\n\n' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)